src/init.cpp

```diff
@@ -1746,7 +1746,10 @@ bool AppInit2()
             uiInterface.InitMessage(_("Rescanning..."));
             LogPrintf("Rescanning last %i blocks (from block %i)...\n", chainActive.Height() - pindexRescan->nHeight, pindexRescan->nHeight);
             const int64_t nWalletRescanTime = GetTimeMillis();
-            pwalletMain->ScanForWalletTransactions(pindexRescan, true);
+            if (pwalletMain->ScanForWalletTransactions(pindexRescan, true, true) == -1) {
+                LogPrintf("Shutdown requested over the txs scan. Exiting.\n");
+                return false;
+            }
```

Comment:
styling nit: `return error("Shutdown requested over the txs scan. Exiting.")` (no need for line terminator `\n`)

Reply:
done